### ISSUE TITLE
Why can’t I resume the tune task normally when using basic variant generator?

### ISSUE THREAD

**Issue by EscapeReality846089495** [ray-project/ray#14003]
Yesterday, I wanted to save my tune task. I tried the resume mechanism mentioned in the manual. As the manual says, checkpoints are persisted at all times. But when I try to terminate the program and restart the tune task with resume as true, it can start normally, but the sampling number is not resumed. When I start in resume mode, ray just rerun the pending or running trails saved in the checkpoint. I found that it was because I used the BasicVariantGenerator which is default choice of search algorithm. It didn’t have any way to reach the checkpoint mechanism so that in the process of resume, the resume method of runner ignores the initialization of some parameters of the search algorithm object(I mean BasicVariantGenerator).
``` python3
        self.__setstate__(runner_state["runner_data"])
        if self._search_alg.has_checkpoint(self._local_checkpoint_dir):
            self._search_alg.restore_from_dir(self._local_checkpoint_dir)
```
I guess the _iteration in the runner_data in the checkpoint file corresponds to the total number of sampling. But it may not be because the _iteration is not always equal to the number of sampling specified in the tune.run method.
To sum up, I can’t normally terminate the tune task while using the basic variant generator and resume it. I can use the search algorithm with checkpoint saving mechanism to avoid this, such as HyperOptSearch that I am using, but is it reasonable that BasicVariantGenerator didn’t reach checkpoint saving mechanism?

**Comment by richardliaw** [comment]
(fixed as discussed on discourse)
